dev/release/utils-create-release-tarball.sh

```diff
@@ -46,18 +46,41 @@ rm -rf ${root_folder}
 # Resolve symbolic and hard links
 rm -rf ${root_folder}.tmp
 mv ${root_folder} ${root_folder}.tmp
-cp -R -L ${root_folder}.tmp ${root_folder}
+cp -a -L ${root_folder}.tmp ${root_folder}
 rm -rf ${root_folder}.tmp
 
 # Create a dummy .git/ directory to download the source files from GitHub with Source Link in C#.
+if stat --help > /dev/null 2>&1; then
+  # GNU stat
+  csharp_mtime=$(date --date="$(stat --format="%y" csharp)" +%Y%m%d%H%M.%S)
```

Comment:
can we add a comment on why is this time necessary? Is csharp somehow generating this file if not there? I can't seem to find where would that fail when building the `tar.gz`

Reply:
Sure. How about https://github.com/apache/arrow/pull/46342/commits/4702c343364b28fa625dfb3cabddf084e36a0547 ?

Reply:
oh! I see we are setting the time, I missed the `-t` on the touch! Thanks @kou that's really good!